File: Report_1/Step_2/Iteration_Or_Recursion/Iteration_Or_Recursion.c
#include <stdio.h>

int recur(int a)
{
	if (a <= 0)
		return a;

	else
		return recur(a - 1) + a;
}

int iter(int a)
{
	int i, sum = 0;

	for (i = 1; i <= a; i++)
		sum += i;

	return sum;
}

int main()
{
	int recursion = recur(10);
	int iteration = iter(10);

	printf("Recursion : %d\n", recursion);
	printf("Iteration : %d\n", iteration);
	// Recursion: 55
	// Iteration : 55

	return 0;
}

File: Report_1/Step_2/Iteration_Or_Recursion/Iteration_Or_Recursion.java


public class Iteration_Or_Recursion {

	static int recu(int n) {
		if (n == 0)
			return 0;

		else
			return n + recu(n - 1);
	}

	public static void main(String[] args) {
		int iter = 0;

		for (int i = 1; i <= 10; i++)
			iter += i;

		System.out.println("Recursion : " + recu(10));
		// Recursion : 55

		System.out.println("Iteration : " + iter);
		// Iteration : 55

	}

}
